lace/archive/nyx_archive.py

```diff
@@ -184,12 +197,60 @@ def _set_info_sim(self):
             "lambda_P",
         ]
 
+    def _get_emu_cosmo(self, nyx_data, sim_label):
+        if self.update_kp == False:
+            emu_cosmo_all = np.load(self.file_cosmo, allow_pickle=True)
+            for ii in range(len(emu_cosmo_all)):
+                if emu_cosmo_all[ii]["sim"] == sim_label:
+                    if emu_cosmo_all[ii]["emu_cosmo"]["kp_Mpc"] != self.kp_Mpc:
+                        self.update_kp = True
+                    else:
+                        cosmo = emu_cosmo_all[ii]["cosmo"]
+                        emu_cosmo = emu_cosmo_all[ii]["emu_cosmo"]
+                    break
+
+        if self.update_kp == True:
+            sim_params = get_attrs(nyx_data[sim_label])
+            if isim == "fiducial":
+                sim_params["A_s"] = 2.10e-9
+                sim_params["n_s"] = 0.966
+                sim_params["h"] = sim_params["H_0"] / 100
+            cosmo = camb_cosmo.get_Nyx_cosmology(sim_params)
+
+            # compute linear power parameters at each z (will call CAMB)
+            linP_zs = fit_linP.get_linP_Mpc_zs(
+                cosmo, self.list_sim_redshifts, self.kp_Mpc
+            )
+            # compute conversion from Mpc to km/s using cosmology
+            dkms_dMpc_zs = camb_cosmo.dkms_dMpc(
+                cosmo, z=np.array(self.list_sim_redshifts)
+            )
+
+            emu_cosmo = {}
+            emu_cosmo["kp_Mpc"] = nyx_archive.kp_Mpc
+            labels = ["Delta2_p", "n_p", "alpha_p", "f_p"]
+            for lab in labels:
+                emu_cosmo[lab] = np.zeros(nz)
+                for ii in range(nz):
+                    emu_cosmo[lab][ii] = linP_zs[ii][lab]
+
+            emu_cosmo["dkms_dMpc"] = np.zeros(nz)
+            emu_cosmo["z"] = np.zeros(nz)
+            for ii in range(nz):
+                emu_cosmo["dkms_dMpc"][ii] = dkms_dMpc_zs[ii]
+                emu_cosmo["z"][ii] = nyx_archive.list_sim_redshifts[ii]
+
+        return cosmo, emu_cosmo
+
     def _load_data(self, file_name, zmax):
         # if file_name was not provided, search for default one
         if file_name is None:
             if "NYX_PATH" not in os.environ:
                 raise ValueError("Define NYX_PATH variable or pass file_name")
             file_name = os.environ["NYX_PATH"] + "/models.hdf5"
+            self.file_cosmo = os.environ["NYX_PATH"] + "/nyx_emu_cosmo.npy"
+        else:
+            self.file_cosmo = os.path.dirname(file_name) + "/nyx_emu_cosmo.npy"
```

Comment:
Is `file_name` a directory name? If so, I would rename the variable to `dir_name` or similar

Reply:
file_name is the name of the file containing the Nyx P1D measurements. What I do in that line is to get the path of that file, as the file nyx_emu_cosmo should also be there